extensions/azure/data-plane/data-factory/src/main/java/org/eclipse/dataspaceconnector/azure/dataplane/azuredatafactory/KeyVaultClient.java

```diff
@@ -0,0 +1,39 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.azure.dataplane.azuredatafactory;
+
+import com.azure.security.keyvault.secrets.SecretClient;
+import com.azure.security.keyvault.secrets.models.KeyVaultSecret;
+
+/**
+ * Client for Azure Key Vault, wrapping the Azure SDK.
+ */
+class KeyVaultClient {
```

Comment:
Why not use `AzureVault`?

Reply:
To reuse the resource-manager module consistently, and because `AzureVault` authenticates with client secret. We could have a follow up task to refactor AzureVault to use resource-manager (enforcing SRP)

Reply:
i'm good with it being a temporary solution, ultimately we should enhance the existing `AzureVault` implementation, and not duplicate functionality